wire-ios-request-strategy/Sources/APIs/MessageAPI.swift

```diff
@@ -58,6 +60,41 @@ class MessageAPIV0: MessageAPI {
         self.httpClient = httpClient
     }
 
+    func broadcastProteusMessage(message: any ProteusMessage) async throws -> (Payload.MessageSendingStatus, ZMTransportResponse) {
+        let path = "/broadcast/otr/messages"
+
+        // FIXME: [jacob] move encryption out of the API WPB-5499
+        guard let encryptedPayload = await message.context.perform({
+            message.encryptForTransport()
+        }) else {
+            WireLogger.messaging.error("failed to encrypt message for transport")
+            throw NetworkError.errorEncodingRequest
+        }
+
+        let request = ZMTransportRequest(
+            path: path,
+            method: .post,
+            binaryData: encryptedPayload.data,
+            type: protobufContentType,
+            contentDisposition: nil,
+            apiVersion: apiVersion.rawValue
+        )
+
+        let response = await httpClient.send(request)
+
+        if response.httpStatus == 412 {
+            guard
+                let messageSendingStatus = Payload.MessageSendingStatusV0(response, decoder: .defaultDecoder)
+            else {
+                throw NetworkError.errorDecodingResponse(response)
+            }
+            throw NetworkError.missingClients(messageSendingStatus.toMessageSendingStatus(domain: ""), response)
```

Comment:
do you need a domain here like `BackendAPI.domain` ?

Reply:
Is `BackendAPI.domain` always set?

Reply:
Last I time looked this would not be necessary, but if I can use `BackendAPI.domain` it would make the code easier to follow.

Anyway I'd prefer to not this change in this PR.

Reply:
ok, not always set as it is optional

Reply:
We should check if can't simply delete API v0 support. AFAIK no one is running a backend which only supports V0